include/alpaka/core/UniformCudaHip.hpp

```diff
@@ -61,7 +61,11 @@ namespace alpaka::uniform_cuda_hip::detail
             if(std::find(std::cbegin(ignoredErrorCodes), std::cend(ignoredErrorCodes), error)
                == std::cend(ignoredErrorCodes))
             {
-                rtCheck<TApi, TThrow>(error, ("'" + std::string(cmd) + "' returned error ").c_str(), file, line);
+                rtCheck<TApi, TThrow>(
+                    error,
+                    (std::string("'") + std::string(cmd) + "' returned error ").c_str(),
```

Comment:
Potentially a dumb question: is the temporary string guaranteed to be valid ?

Isn't it safer to write
```c++
{
  std::string msg = std::string("'") + std::string(cmd) + "' returned error ";
  rtCheck<TApi, TThrow>(error, msg, file, line);
}
```
?

Reply:
Also
```c++
'"'s + std::string(cmd) + "' returned error "s
```
would look nicer :-)

Can we add
```c++
using namespace std::literals;
```
inside the `namespace alpaka::uniform_cuda_hip::detail` namespace ?

Reply:
The lifetime of a temporary ends at the end of the expression, so practically it lives until the next `;`: "All temporary objects are destroyed as the last step in evaluating the [full-expression](https://en.cppreference.com/w/cpp/language/expressions#Full-expressions) that (lexically) contains the point where they were created" (from [cppreference](https://en.cppreference.com/w/cpp/language/lifetime)).